libflux/flux-core/src/parser/mod.rs

```diff
@@ -16,9 +16,9 @@ mod strconv;
 ///
 /// Returns a [`File`] with the value of the `name` parameter
 /// as the file name.
-pub fn parse_string(name: &str, s: &str) -> File {
+pub fn parse_string(name: String, s: &str) -> File {
```

Comment:
Is there a reason for this change? It seems to me like the old function signature was more correct.

Reply:
The previous implementation made a clone of the name. My general philosophy in Rust is that if a function needs to own the data it should state it in the function signature. This allows the called to decide best how to make a copy or otherwise control the allocations. Otherwise calling parse_string would always require a new allocation.